File: Bigdata_eunseok/java/workspace/12-Class/src/Exam4.java
class Calc{
	int plus(int x, int y) {
		return x + y;
	}
	int minus(int x, int y) {
		return x - y;
	}
	int multiply(int x, int y) {
		return x * y;
	}
	int divide(int x, int y) {
		return x / y;
	}
	int f(int x, int y) {
		return plus(x, y) + multiply(x, y);
	}
}
public class Exam4 {
	public static void main(String[] args) {
		Calc calc = new Calc();
		System.out.println(calc.plus(5, 7));
		System.out.println(calc.minus(5, 7));
		System.out.println(calc.multiply(5, 7));
		System.out.println(calc.divide(5, 7));
		System.out.println(calc.f(5, 7));
	}
}

File: Bigdata_eunseok/python/workspace/14-class/exam4.py
# -*- coding: utf-8 -*-
"""
Created on Thu Sep 16 16:37:18 2021

@author: eunse
"""

def plus1(x, y):
    print("함수 plus() 호출")
    return x + y
    
class Calc:
    #@staticmethod
    def plus2(x, y):
        print("스테틱 plus() 호출")
        return x + y
    
    def plus(self, x, y):
        print("메소드 plus() 호출")
        return x + y
    
    def minus(self, x, y):
        print("메소드 minus() 호출")
        return x - y
    
    def f(self, x, y):
        result1 = plus1(x, y)
        result2 = self.plus(x, y)
        result3 = Calc.plus2(x, y)
        result4 = result1 + result2 + result3
        return result4
    
print(Calc.plus2(5, 7))

cc = Calc()

print(cc.plus(5, 7))

print(cc.f(5, 7))